<a href="https://colab.research.google.com/github/riraonline/data_science_2026/blob/main/Pertemuan2_Richardin_Rafsanjani_220401010116.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
import numpy as np
import pandas as pd

In [5]:
url = "https://raw.githubusercontent.com/datasciencedojo/datasets/refs/heads/master/titanic.csv"
df = pd.read_csv(url)

In [6]:
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


**Jawaban Soal**

**1. Berapa jumlah baris dan kolom pada dataset Titanic? Kolom apa saja yang tersedia?**

In [8]:
df.shape
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

**Jawaban:**

*   **Jumlah baris:** 891
*   **Jumlah kolom:** 12

**Nama Kolom:** PassengerId, Survived, Pclass, Name, Sex, Age,
SibSp, Parch, Ticket, Fare, Cabin, Embarked



**2. Kolom mana saja yang memiliki missing values? Berapa jumlah dan proporsinya (%)
per kolom?**

In [9]:
missing = df.isnull().sum()
percent = (missing / len(df)) * 100

pd.DataFrame({
    'Missing': missing,
    'Percent (%)': percent
})

,Missing,Percent (%)
PassengerId,0,0.000000
Survived,0,0.000000
Pclass,0,0.000000
Name,0,0.000000
Sex,0,0.000000
Age,177,19.865320
SibSp,0,0.000000
Parch,0,0.000000
Ticket,0,0.000000
Fare,0,0.000000


**Jawaban:**

Kolom yang bermasalah utama:

*   Age - 177 - 19,9%
*   Cabin - 687 - 77,1%
*   Embarked - 2 - 0,22%





**3. Berapa persen penumpang yang selamat (Survived=1)? Gunakan operasi NumPy
atau Pandas.**

In [10]:
survival_rate = df['Survived'].mean() * 100
survival_rate

np.float64(38.38383838383838)

**Jawaban:**

Mayoritas penumpang sebanyak 38.38%, sementara sisa nya tidak selamat sebanyak ±61%

**4. Filter penumpang wanita dari kelas 1 (Pclass=1). Berapa jumlahnya dan berapa rata-
rata usianya?**

In [11]:
female_p1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)]

count = female_p1.shape[0]
avg_age = female_p1['Age'].mean()

count, avg_age

(94, np.float64(34.61176470588235))

**Jawaban:**

*   Jumlah: 94 orang
*   Rata-rata usia: ±34,6 tahun



**5. Gunakan groupby untuk menghitung tingkat keselamatan (mean Survived) per kelas
penumpang. Kelas mana yang paling tinggi?**

In [12]:
survival_by_class = df.groupby('Pclass')['Survived'].mean()
survival_by_class

,Survived
Pclass,
1,0.629630
2,0.472826
3,0.242363


**Jawaban:**

Kelas 1 cukup di pastikan selamat dan hal ini bisa menunjukkan bahwa semakin kaya seseorang maka semakin tinggi peluang selamat nya

**6. Buat analisis tambahan berdasarkan variabel lain yang Anda anggap menarik.
Sertakan interpretasi singkat.**

In [13]:
df.groupby('Sex')['Survived'].mean()

,Survived
Sex,
female,0.742038
male,0.188908


**Jawaban:**

Analisi tambahan, yang mengatakan bahwa gender atau jenis kelamin perempuan lebih tinggi tingkat keselamatan nya.